api/controller/dbController.js

```diff
@@ -0,0 +1,94 @@
+import { query } from '../../db/db'
+import { 
+  artistQueries, setlistQueries, songQueries, setlistSongQueries
+  } from '../../db/queries'
+
+  export class dbController {
+    artistId;
+    concertId;
+    songId;
+    constructor(apiResult, artist, song, artistId, concertId, songId) {
+      this.apiResult = apiResult;
+      this.artist = artist;
+      this.song = song;
+      this.externalId = apiResult.setlist[0].artist.mbid;
+      this.artistId = artistId;
+      this.concertId = concertId;
+      this.songId = songId;
+    }
+
+    addSong = async (song) => {
+      const addResult = await query(songQueries.addSong,[song,this.artistId]);
+      this.songId = addResult[0].id;
+    }
+    
+    getSong = async (song) => {
+      const getResult = await query(songQueries.getSong,[song,this.artistId]);
+      const songInDb = getResult.length > 0;
+      if (songInDb) {
+        this.songId = getResult[0].id;
+      } else {
+        this.songId = null;
+      }
+    }
+
+    addSetlistSong = async () => {
+      await query(setlistSongQueries.addSetlistSong,[this.concertId,this.songId])
+    }
+
+    addSetlist = async (externalId,updatedAt,eventDate) => {
+      const result = await query(setlistQueries.addSetlist,[externalId,this.artistId,updatedAt,eventDate]);
+      return result[0].id;
+    }
+  
+    addedArtist = async () => {
+      const result = await query(artistQueries.addArtist,[this.externalId,this.artist]);
+      this.artistId = result[0].id;
+    }
+  
+    getArtist = async () => {
+      const result = await query(artistQueries.getArtist,[this.artist]);
+      this.artistId = result.length > [0] ? result[0].id : 0;
+      return this.artistId;
+    }
+
+    getDates = async () => {
+      const returnedValue = {};
+      const result = await query(setlistSongQueries.getSetlistSong,[this.song,this.artistId])
+      const compiledDates = this.compileDates(result);
+      returnedValue[this.song] = compiledDates;
+      return returnedValue;
+    }
+    
+    compileSongs = (sets) => {
+      const songsFromThisConcert = [];
+      sets.forEach(set => set.song.forEach(song => songsFromThisConcert.push(song.name.toLowerCase())));
+      return songsFromThisConcert;
+    }
+    compileDates = (arrayofDateObjects) => {
+      const finalDateArray = [];
+      arrayofDateObjects.forEach(dateObject => finalDateArray.push(dateObject.date));
+      return finalDateArray;
+    }
```

Comment:
is putDatesInOneArray too verbose?

Reply:
I think it would be too verbose. 

The specificity of a name depends on the scope of the variable. Global variables should be very verbose, while variables defined within one-line functions can be just a letter, like `array.map(d => d.date)`. Because this is a private method that's just being used for this file I think it's not too bad if it's not particularly descriptive. If we want something more descriptive than `getDates` maybe `datesFromDateObjects` could work?

Reply:
As an aside, you generally don't need to specify whether something is an array in the name of the variable. If the name is plural it implies that the object is an array. If you named the argument `dateObjects` I would assume it's an array of date objects. Because of this useful paradigm, prepending `arrayOf` is a little redundant.